rdr_service/dao/study_nph_dao.py

```diff
@@ -506,54 +487,77 @@ def _is_order_cancelled(order: Order) -> bool:
     def _is_ordered_sample_cancelled(ordered_sample: OrderedSample) -> bool:
         return str(ordered_sample.status).lower() == "cancelled"
 
-    def get_biospecimens_for_order(self, nph_participant: Participant, order: Order) -> Iterator[Dict[str, Any]]:
+    def _get_biospecimens_for_order(
+        self, nph_participant: Participant, order: Order, ordered_samples: Iterable[OrderedSample]
+    ) -> Iterator[Dict[str, Any]]:
         nph_stored_sample_session = NphStoredSampleDao()
         nph_study_category_dao = NphStudyCategoryDao()
-        with self.session() as session:
-            ordered_samples_for_participant: Iterable[OrderedSample] = list(
-                session.query(OrderedSample).filter(OrderedSample.order_id == order.id).all()
+        for ordered_sample in ordered_samples:
+            parent_study_category = nph_study_category_dao.get_parent_study_category(order.category_id)
+            nph_module_id = nph_study_category_dao.get_parent_study_category(parent_study_category.id)
+            sample_processing_ts = ordered_sample.collected if not ordered_sample.parent is None else None
```

Comment:
suggesting change for readability and conformance to pep8

```suggestion
            sample_processing_ts = ordered_sample.collected if ordered_sample.parent is not None else None
```